src/libAtomVM/nifs.c

```diff
@@ -1284,14 +1284,42 @@ static term nif_erlang_spawn(Context *ctx, int argc, term argv[])
 static term nif_erlang_send_2(Context *ctx, int argc, term argv[])
 {
     UNUSED(argc);
+    term target = argv[0];
+    GlobalContext *glb = ctx->global;
+    struct ListHead *processes_table_list = synclist_rdlock(&glb->processes_table);
+
+    if (term_is_pid(target)) {
+        int32_t target_pid = term_to_local_process_id(target);
+
+        globalcontext_send_message_nolock(glb, target_pid, argv[1]);
+        synclist_unlock(&glb->processes_table);
```

Comment:
Why we need to lock and unlock here when we have a `globalcontext_send_message` function?

Reply:
See Paul's comments comments in the resolved conversation around line 1294 in `nifs.c`.

Reply:
I have changed my conversation with Paul about this to "unresolved" just to make it easier to find.

Reply:
for clarity I would this in a different way:
- pid branch: let's keep code as it is, no need to lock the process table, regular send will do that for us
- atom branch: let's move here the lock - get pid - send_no_lock - unlock + let's add a comment before locking that explains why

By doing this I think we get an easier to understand flow.